### PR TITLE
Provide destination and endpoint while requesting for reports

### PR DESCRIPTION
**Pull request by pan-apple**
#### Problem
Reporting should not require a client to setup a binding table entry.

#### Change overview
Add target destination (node and endpoint) to `ConfigureAttribute` commands. This command is used for turning on/off reporting for an attribute. The node generating the report will use the target information to dispatch the reports.

#### Testing
Removed the binding steps from the current unit test, and ensure it continues to work (receive reports).

### CODE REVIEW

File: src/app/reporting/reporting.cpp [hunk 1]
```diff
@@ -66,7 +66,13 @@ using namespace chip;
 
 #define NULL_INDEX 0xFF
 
-static void conditionallySendReport(EndpointId endpoint, ClusterId clusterId);
+/*
+ * This method sends the report to the provided destination node at destinationEndpoint.
+ * If the reportDestination is kUndefinedNodeId, the method uses binding table to identify
+ * destination nodes and endpoints
+ */
+static void conditionallySendReport(EndpointId endpoint, ClusterId clusterId, NodeId reportDestination,
```
Comment (kghost) [inline src/app/reporting/reporting.cpp:74]: According to the comment, It is confusing that the code works this way. Can it take a `MessageSendDestination` argument instead of `NodeId reportDestination`

File: src/app/reporting/reporting.cpp [hunk 2]
```diff
@@ -259,15 +268,17 @@ void emberAfPluginReportingTickEventHandler(void)
             {
                 emberAfReportingPrintln("Reporting Entry Full - creating new report");
             }
-            conditionallySendReport(apsFrame->sourceEndpoint, apsFrame->clusterId);
+            conditionallySendReport(apsFrame->sourceEndpoint, apsFrame->clusterId, reportDestination, destinationEndpoint);
```
Comment (kghost) [inline src/app/reporting/reporting.cpp:271]: If we have implemented `emberAfGetDestinationByBinding`, here it can be:

```
conditionallySendReport(apsFrame->sourceEndpoint, apsFrame->clusterId, emberAfGetDestinationByBinding(apsFrame), destinationEndpoint);
```

File: src/app/reporting/reporting.cpp [hunk 3]
```diff
@@ -354,17 +365,27 @@ void emberAfPluginReportingTickEventHandler(void)
 
     if (apsFrame != NULL)
     {
-        conditionallySendReport(apsFrame->sourceEndpoint, apsFrame->clusterId);
+        conditionallySendReport(apsFrame->sourceEndpoint, apsFrame->clusterId, reportDestination, destinationEndpoint);
```
Comment (kghost) [inline src/app/reporting/reporting.cpp:368]: I'm a little confused about why there is another `if (apsFrame != NULL)` test, is it able to move this statement inside `if (apsFrame == NULL)` branch ?

**Comment by andy31415** [comment]
We are having discussion required regarding the PRs for subscriptions regarding usage or not usage of reporting.

Marking this discussion required: I assume this is the PR (or part of the PR) that will be considered for a proposal for subscription implementation.

**Review by andy31415 — CHANGES_REQUESTED** [review]
Marking requested changes: please provide a complete PR for anything related to subscriptions, so we can compare approaches.

**Comment by woody-apple** [comment]
We should discuss and next week’s meeting about how to proceed here, however we’ll need to timebox this so as to unblock subscription work.

**Review by woody-apple — APPROVED** [review]

File: src/app/reporting/reporting.cpp (hunk 1 above)
Comment (woody-apple) [reply src/app/reporting/reporting.cpp:74]: @pan-apple?

**Comment by woody-apple** [comment]
Per discussion today, closing this PR - given this is now covered here: https://github.com/project-chip/connectedhomeip/pull/9510
